06-classification_continued.Rmd

```diff
@@ -559,16 +559,257 @@ The overall workflow for performing K-nearest neighbour classification using `ti
 
 **Strengths:**
 
-1. Simple and easy to understand
-2. No assumptions about what the data must look like 
-3. Works easily for binary (two-class) and multi-class (> 2 classes) classification problems
+1. Simple, intuitive algorithm
+2. Few assumptions about what the data must look like 
+3. Works for binary (two-class) and multi-class (> 2 classes) classification problems
 
 **Weaknesses:**
 
-1. As data gets bigger and bigger, K-nearest neighbour gets slower and slower, quite quickly
-2. Does not perform well with a large number of predictors
+1. As the training data gets larger, K-nearest neighbours becomes very slow
+2. May not perform well with a large number of predictors
 3. Does not perform well when classes are imbalanced (when many more observations are in one of the classes compared to the others)
 
+## Predictor variable selection
+
+> This section is not required reading for the remainder of the textbook.
+
+Another potentially important part of tuning your classifier is to choose which
+variables from your data will be treated as predictor variables. Based
+on your initial exploratory analysis, you can choose
+anything from using a single predictor variable to using every variable in your
+data; the K-nearest neighbours algorithm accepts any number of
+predictors. However, it is **not** the case that using more predictors always
+yields better predictions! In fact, including irrelevant predictors can
+actually negatively affect classifier performance.
+
+Let's take a look at an example where K-nearest neighbours performs
+worse when given more predictors to work with. Here we have modified
+the breast cancer data to include the variables `Smoothness` and `Concavity`,
+as well as 500 irrelevant variables
+named `Irrelevant1` to `Irrelevant500`, which
+we generated ourselves using a random number generator.
+You will see from the scatter plot below with two example
+variables `Irrelevant1`, `Irrelevant2` that
+these irrelevant variables have no meaningful relationship
+with the `Class` response variable.
```

Comment:
Here you use response variable, but earlier in the chapter we use target. I think we need to come up with a chosen terminology for this book, and use it consistently. When we first introduce the term, we can qualify the other terms that also mean the same thing. We might want to consider making a dictionary, like we have in MDS: https://ubc-mds.github.io/resources_pages/terminology/ (search for "Equivalence classes").

Reply:
i agree -- that is part of a good "index" (and why it's really hard to automate index generation). When we finally do formatting stuff and I generate the index, I'll make sure to list all of the versions of these.

I've changed everything to "response" for now, but I have also opened an issue to add "other common words people use" at the point of first introduction as part of the specific chapter revisions.

Reply:
Issue #165 for this